draft-normington-jsonpath-latest.xml

```diff
@@ -384,50 +384,115 @@ array-slice = [ integer ] ws %x3A ws [ integer ]
                   </section>
                   <section numbered="false" toc="exclude">
                      <name>Semantics</name>
-                     <t>An array index selects any element of an array which is indexed by the integer index.
-                        It selects no nodes from a node which is not an array.
-                     </t>
-                     <t>A non-negative index <tt>i</tt> selects an element of an array of length <tt>len</tt> if and only if <tt>0 &lt;= i &lt; len</tt>.
-                        The index <tt>0</tt> selects the first element of a non-empty array and, providing <tt>i &lt; len - 1</tt>, index <tt>i + 1</tt>
-                        selects the element immedately after (in array order) the element selected by index <tt>i</tt>.
-                     </t>
-                     <t>A negative index <tt>j</tt> selects an element of an array of length <tt>len</tt> if and only if <tt>0 &lt;= j + len &lt; len</tt>,
-                        in which case it selects the same element as the non-negative index <tt>j + len</tt>.
-                     </t>
-                     <t>An array slice selects any elements of an array which are indexed by the members of the slice,
-                        as defined below, in the same order as the slice.
-                        It selects no nodes from a node which is not an array.
-                     </t>
-                     <t>The members of a slice depend on whether the step value is positive, negative, or zero as follows. If the step value is
-                        omitted, it defaults to <tt>1</tt>.
-                     </t>
-                     <t>For positive integer <tt>step</tt> and integers <tt>start</tt> and <tt>end</tt>, the array slice
-                        <tt>start:end:step</tt> consists of the indices starting at <tt>start</tt> and increasing repeatedly
-                        by <tt>step</tt> while the resultant index is less than <tt>end</tt>.
-                        The indices of the slice are the same as the values of i iterated over by this <tt>for</tt> loop in C:
-                     </t>
-                     <sourcecode type="c">
-for (i = start; i &lt; end; i = i + step) {
-  ...
-}
+                     <t>Array slicing is inspired by the behaviour of the <tt>Array.prototype.slice</tt> method
+                        of the JavaScript language as defined by ECMA-262, with the addition of the <tt>step</tt>
```

Comment:
Probably worth adding a non-normative reference for ECMA-262.

Reply:
added